File: frontend/src/types/curriculum.ts
export type CurriculumStatus = 'PENDING' | 'APPROVED' | 'REJECTED'

export interface ApplicationInput {
  vacancyId: string | null
  applicantName: string
  applicantEmail: string
  applicantPhone: string | null
  language: string
  file: File | null
  certificates: File[]
}

export interface CertificateFile {
  id: string
  fileName: string
  fileUrl: string
}

export interface Curriculum {
  id: string
  vacancyId: string | null
  applicantName: string
  applicantEmail: string
  applicantPhone: string | null
  fileName: string | null
  fileUrl: string | null
  certificates: CertificateFile[]
  submittedAt: string
  status: CurriculumStatus
}

File: backend/src/main/java/taller/multimedia/backend/model/curriculum/Curriculum.java
package taller.multimedia.backend.model.curriculum;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;
import org.hibernate.annotations.BatchSize;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "curriculums")
@Data
public class Curriculum {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    // Sin relación JPA a Vacancy a propósito: se guarda solo el id, igual
    // que Vacancy.filledByApplicationId. Evita acoplar las dos entidades.
    @Column(name = "vacancy_id")
    private UUID vacancyId;

    @Column(name = "applicant_name", nullable = false)
    private String applicantName;

    @Column(name = "applicant_email", nullable = false)
    private String applicantEmail;

    @Column(name = "applicant_phone")
    private String applicantPhone;

    @Column(name = "file_name")
    private String fileName;

    @Column(name = "file_url")
    private String fileUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CurriculumStatus status = CurriculumStatus.PENDING;

    @Column(name = "submitted_at", nullable = false, updatable = false)
    private LocalDateTime submittedAt = LocalDateTime.now();

    @Column(name = "language")
    private String language;

    @OneToMany(mappedBy = "curriculum", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @BatchSize(size = 20)
    private List<CurriculumCertificate> certificates = new ArrayList<>();
}
